v-next/hardhat/src/internal/builtin-plugins/network-manager/edr/errors.ts

```diff
@@ -0,0 +1,131 @@
+import type { ProviderRpcError } from "../../../../types/providers.js";
+
+import { CustomError } from "@ignored/hardhat-vnext-utils/error";
+
+// Codes taken from: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1474.md#error-codes
+//
+// Code	  Message	              Meaning	                            Category
+//
+// -32700	Parse error	          Invalid JSON	                      standard
+// -32600	Invalid request	      JSON is not a valid request object  standard
+// -32601	Method not found	    Method does not exist	              standard
+// -32602	Invalid params	      Invalid method parameters	          standard
+// -32603	Internal error	      Internal JSON-RPC error	            standard
+// -32004	Method not supported	Method is not implemented	          non-standard
+// -32000	Invalid input	        Missing or invalid parameters	      non-standard
+// -32003	Transaction rejected	Transaction creation failed	        non-standard
+//
+//
+// Non standard:
+
+// -32999 Invalid response      The server returned a JSON-RPC      hardhat-sepecific
+//                              response, but the result is not
+//                              in the expected format
+//
+// Not implemented:
+//
+// -32001	Resource not found	  Requested resource not found	      non-standard
+// -32002	Resource unavailable	Requested resource not available	  non-standard
```

Comment:
This file is taken over directly.

Some errors in here are not used. I kept them for completeness.